libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Network/Network/Action/CSFSalesforceAction.m

```diff
@@ -86,7 +86,7 @@ - (void)setEnqueuedNetwork:(CSFNetwork *) network {
         if (_enqueuedNetwork) {
             if ([self shouldUpdateBaseUrl]) {
                 self.baseURL = self.enqueuedNetwork.account.credentials.apiUrl;
-                self.cachedAPIURL = self.enqueuedNetwork.account.credentials.apiUrl;
+                self.cachedAPIURL = self.baseURL;
```

Comment:
setBaseURL is overridden in CSFAction, and it has some logic for adding a trailing slash to the url. If that happens, then baseURL and cachedAPIURL would not be equal. That's why it's better to set cachedAPIURL directly to self.baseURL.